Makefile.am

```diff
@@ -192,7 +192,11 @@ LCOV_FILTER_PATTERN = \
 	-p "src/secp256k1" \
 	-p "depends"
 
-baseline.info:
+$(COV_TOOL_WRAPPER):
+	@echo 'exec $(COV_TOOL) "$$@"' > $(COV_TOOL_WRAPPER)
```

Comment:
Could add a comment that this is needed because of a potential dash in the filename of COV_TOOL? (At least I think that was the reason a wrapper is required)

Reply:
My understanding is that the wrapper is needed because we run `lcov --gcov-tool /path/to/gcov` and our gcov is `llvm-cov gcov`, so `lcov --gcov-tool "llvm-cov gcov" other arguments` would not work.

It is also possible to generate the coverage without using gcov/lcov:

```sh
... compile with clang++ -fprofile-instr-generate -fcoverage-mapping ...
export LLVM_PROFILE_FILE="/tmp/coverage/%m-%p.profraw"
... run unit + functional tests ...
llvm-profdata merge /tmp/coverage/*.profraw -o all.profdata

# Generate lcov-like html report:
llvm-cov export -instr-profile=all.profdata -format=lcov src/test/test_bitcoin -object=src/bitcoind > coverage_lcov.info
genhtml --output-directory /tmp/coverage/html --legend --branch-coverage coverage_lcov.info

# Or generate llvm html report (better templates visualization!):
llvm-cov show -instr-profile=all.profdata -format=html --output-dir=/tmp/coverage/html src/test/test_bitcoin -object=src/bitcoind
```